### DISCUSSION TITLE
Discrepancy external GPX and GoPro

### DISCUSSION THREAD

**Discussion by rmoriz** [time4tea/gopro-dashboard-overlay#202] [Q&A]
Hello,

awesome software! I've just one minor thing to nitpick, maybe I'm doing it wrong or missed and option?

Scenario "bicycle":
- GoPro recording video+audio, gps.
- bicycle computer (Wahoo) recording to gpx/fit with external sensors connected (e.g. powermeter, cadence, heart rate)

Both are a little bit out of sync, maybe 0.5-1.5s? I'm pretty sure it's caused by the latency of the sensor data arriving at the bike computer. 

Is there an option to configure a (negative) time "gap" between video an gpx/fit? like "-1s" but still using the automatic matching?

Thank you!

**Comment by time4tea** [comment]
Hi, there's not currently an option to configure an offset.

At the moment we use the time from the video to try to sync everything together, as this is very accurately recorded in the gopro file.

With most bike gps only giving one value per second, it gets a bit tricky to know what they are reporting.. is it the instantaneous value at the point, the average for that second, or some mix of the two?

At the moment we align the values by gps time, but maybe this is a wrong approach.. perhaps it should be the mid point...

Anyway, it's a good suggestion, thanks!!

**Comment by bphilli7** [comment]
I have this same setup for my biking. A gopro hero10 and a wahoo elemnt bolt v2.

Even though hero10 and elemnt bolt v2 both have GPS receivers, they do not report the same timestamps in their files. That's confusing to me, as they both have ability to sync to GPS time.

Running a recent test with my cell phone, gopro, and element, I measured the time offset between the three devices.

The cell phone time was 1.4 seconds delayed from the gopro hero 10 timestamp. The wahoo elemnt bolt v2 was a further 1 second delayed from the cell phone (2.4 seconds from the hero 10).

**Comment by bphilli7** [comment]
I believe this simple update will shift the GPX data +/- some number of seconds to account for an offset between the GoPro timestamps and GPX/FIT computer timestamps.

[gpx-timeshift.patch](https://github.com/user-attachments/files/16517544/gpx-timeshift.patch)

**Reply by iliasalk** [reply]
I can verify this works great!
